_aggregations/metric/scripted-metric.md

```diff
@@ -9,44 +9,220 @@ redirect_from:
+
+## Using parameters in scripts
+
+You can optionally pass custom parameters to your scripts using the `params` field. This is a user-defined object whose contents become variables available in your `init_script`, `map_script`, and `combine_script`. The `reduce_script` does not directly receive `params`, because by the `reduce` phase you all needed data should be in the `states` array. If you need a constant in the `reduce` phase, you can include it as part of each shard’s state or use a stored script. All parameters must be defined inside the global `params` object, this ensures that they are shared across the different script stages. If you do not specify any `params`, it will default to an empty object. 
+
+For example, you can supply a `threshold` or `field` name via `params` and reference `params.threshold` or `params.field` inside your scripts, see following example:
 
 ```json
-GET opensearch_dashboards_sample_data_logs/_search
+"scripted_metric": {
+  "params": {
+    "threshold": 100,
+    "field": "amount"
+  },
+  "init_script": "...",
+  "map_script": "...",
+  "combine_script": "...",
+  "reduce_script": "..."
+}
+```
+
+## Example
+
+The following are examples
+
+### Calculating net profit from transactions
+
+The following example demonstrates the use of the `scripted_metric` aggregation to compute a custom metric that is not directly supported by built-in aggregations. The dataset represents financial transactions, where each document is classified as either a sale (income) or a cost (expense) and includes an amount field. The objective is to calculate the total net profit by subtracting the total cost from the total sales across all documents. 
+
+Create an index:
+
+```json
+PUT transactions
 {
-  "size": 0,
+  "mappings": {
+    "properties": {
+      "type":   { "type": "keyword" }, 
+      "amount": { "type": "double" }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Index four transactions, two sales (amounts `80` and `130`) and two costs (`10` and `30`):
+
+```json
+PUT transactions/_bulk?refresh=true
+{ "index": {} }
+{ "type": "sale", "amount": 80 }
+{ "index": {} }
+{ "type": "cost", "amount": 10 }
+{ "index": {} }
+{ "type": "cost", "amount": 30 }
+{ "index": {} }
+{ "type": "sale", "amount": 130 }
+```
+{% include copy-curl.html %}
+
+In order to run a search with a scripted metric aggregation to calculate the profit, use the following scripts:
+
+- The `init_script` sets up an empty list to hold transaction values for each shard. 
+- The `map_script` for each document adds the document’s amount to the `state.transactions` list, as a positive value if type is `sale`, or as a negative value if type is `cost`. By the end of the `map` phase, each shard’s `state.transactions` will contain a list of numbers representing the income and expenses on that shard. 
+- The `combine_script` takes that list and sums it up to produce the shard’s total profit and returns that number. 
```

Comment:
~that list~ ->  mention the list name received from above

~that number~ -> mention the object name which will be passed onto reduce stage next